library/core/src/slice/ascii.rs

```diff
@@ -55,6 +59,96 @@ impl [u8] {
             byte.make_ascii_lowercase();
         }
     }
+
+    /// Slice version of [`u8::escape_ascii`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(inherent_ascii_escape)]
+    ///
+    /// let s = b"0\t\r\n'\"\\\x9d";
+    /// let escaped = s.escape_ascii().to_string();
+    /// assert_eq!(escaped, "0\\t\\r\\n\\'\\\"\\\\\\x9d");
+    /// ```
+    #[unstable(feature = "inherent_ascii_escape", issue = "77174")]
+    pub fn escape_ascii(&self) -> EscapeAscii<'_> {
+        EscapeAscii { inner: self.iter().flat_map(EscapeByte) }
+    }
+}
+
+impl_fn_for_zst! {
+    #[derive(Clone)]
+    struct EscapeByte impl Fn = |byte: &u8| -> ascii::EscapeDefault {
+        ascii::escape_default(*byte)
+    };
+}
+
+/// An iterator over the escaped version of a byte slice.
+///
+/// This `struct` is created by the [`escape_ascii`] method. See its
+/// documentation for more.
+///
+/// [`escape_ascii`]: ../../std/primitive.slice.html#method.escape_ascii
```

Comment:
I think you might want to disambiguate what type `escape_ascii` is on here:

```suggestion
/// This `struct` is created by the [`[u8]::escape_ascii`] method. See its
/// documentation for more.
///
/// [`[u8]::escape_ascii`]: ../../std/primitive.slice.html#method.escape_ascii
```